server/planning/planning_article_export.py

```diff
@@ -72,9 +74,50 @@ def group_items_by_agenda(items):
                 if agenda is not None and agenda['is_enabled']:
                     agenda['items'] = [item]
                     agendas.append(agenda)
+
+    # replace each agenda id with the actual object
+    for item in items:
+        item_agendas_ids = item.get('agendas', [])
+        item_agendas = []
+        for agenda_id in item_agendas_ids:
+            agenda_in_array = [a for a in agendas if a['_id'] == agenda_id]
+            if len(agenda_in_array) > 0:
+                item_agendas.append(agenda_in_array[0])
+        item['agendas'] = item_agendas
+
     return agendas
 
 
+def inject_internal_converages(items):
+    items_injected = []
+
+    coverage_labels = {}
+    cv = get_resource_service('vocabularies').find_one(req=None, _id='g2_content_type')
+    if cv:
+        coverage_labels= {_type['qcode']: _type['name'] for _type in cv['items']}
+
+    for item in items:
+        if item.get('coverages'):
+            item['internal_coverages'] = []
+            for coverage in item.get('coverages'):
+                user = None
+                assigned_to = coverage.get('assigned_to') or {}
+
+                if assigned_to.get('coverage_provider'):
+                    user = assigned_to['coverage_provider']
+                elif assigned_to.get('user'):
+                    user = get_resource_service('users').find_one(req=None, _id=assigned_to.get('user'))
+
+                if user is not None:
+                    coverage_type = coverage.get('planning').get('g2_content_type')
+                    label = coverage_labels.get(coverage_type, coverage_type)
+                    item['internal_coverages'].append({ "user": user, "type": label })
+
+        items_injected.append(item)
```

Comment:
Here you're creating a new array and adding every single item into this new array and returning that.

It might be better to modify the passed in `items` argument instead.

Reply:
@MarkLark86 right, I was aware of that. I intended to make a "pure" function where the arguments are not modified, but I guess that's not python friendly :)

I'll change it 👍